py/torch_tensorrt/dynamo/Debugger.py

```diff
@@ -0,0 +1,171 @@
+import os
+import tempfile
+from logging.config import dictConfig
+from typing import Any, List, Optional
+
+import torch
+from torch_tensorrt.dynamo.lowering import (
+    ATEN_POST_LOWERING_PASSES,
+    ATEN_PRE_LOWERING_PASSES,
+)
+
+_LOGGER = logging.getLogger("torch_tensorrt [TensorRT Conversion Context]")
+GRAPH_LEVEL = 5
+logging.addLevelName(GRAPH_LEVEL, "GRAPHS")
+
+
+class Debugger:
+    def __init__(
+        self,
+        log_level: str,
+        capture_fx_graph_before: Optional[List[str]] = None,
+        capture_fx_graph_after: Optional[List[str]] = None,
+        save_engine_profile: bool = False,
+        logging_dir: Optional[str] = None,
+    ):
+        self.debug_file_dir = tempfile.TemporaryDirectory().name
+        if log_level != "graphs" and (capture_fx_graph_after or save_engine_profile):
+            _LOGGER.warning(
+                "Capture FX Graph or Draw Engine Graph is only supported when level is 'graphs'"
+            )
+
+        if log_level == "debug":
+            self.log_level = logging.DEBUG
+        elif log_level == "info":
+            self.log_level = logging.INFO
+        elif log_level == "warning":
+            self.log_level = logging.WARNING
+        elif log_level == "error":
+            self.log_level = logging.ERROR
+        elif log_level == "internal_errors":
+            self.log_level = logging.CRITICAL
+        elif log_level == "graphs":
+            self.log_level = GRAPH_LEVEL
+
+        else:
+            raise ValueError(
+                f"Invalid level: {log_level}, allowed levels are: debug, info, warning, error, internal_errors, graphs"
+            )
+
+        self.capture_fx_graph_before = capture_fx_graph_before
+        self.capture_fx_graph_after = capture_fx_graph_after
+
+        if logging_dir is not None:
+            self.debug_file_dir = logging_dir
+        os.makedirs(self.debug_file_dir, exist_ok=True)
+
+    def __enter__(self) -> None:
+        self.original_lvl = _LOGGER.getEffectiveLevel()
+        self.rt_level = torch.ops.tensorrt.get_logging_level()
+        dictConfig(self.get_config())
+
+        if self.log_level == GRAPH_LEVEL:
+            self.old_pre_passes, self.old_post_passes = (
+                ATEN_PRE_LOWERING_PASSES.passes,
+                ATEN_POST_LOWERING_PASSES.passes,
+            )
+            pre_pass_names = [p.__name__ for p in self.old_pre_passes]
+            post_pass_names = [p.__name__ for p in self.old_post_passes]
+            path = os.path.join(self.debug_file_dir, "lowering_passes_visualization")
+            if self.capture_fx_graph_before is not None:
+                pre_vis_passes = [
+                    p for p in self.capture_fx_graph_before if p in pre_pass_names
+                ]
+                post_vis_passes = [
+                    p for p in self.capture_fx_graph_before if p in post_pass_names
+                ]
+                ATEN_PRE_LOWERING_PASSES.insert_debug_pass_before(pre_vis_passes, path)
```

Comment:
The insert debug pass apis already handle a list of pass names and validates them. Not sure if you need to do the validation again here.

Reply:
I think the error on invalid pass is fine. Not sure why we error there but filter here. We should just error and make it consistent

Reply:
So here when we are splitting the list into pre or post if there is a pass that doesnt match either it should be an error I think